cdap-app-fabric/src/main/java/co/cask/cdap/config/PreferencesStore.java

```diff
@@ -153,4 +185,43 @@ private String getMultipartKey(String... parts) {
     }
     return Bytes.toString(result);
   }
+
+  /**
+   * Validate the profile status is enabled and set the preferences in same transaction
+   */
+  private void validateAndSetPreferences(String namespace,
+                                         Config config) throws NotFoundException, ProfileConflictException {
+    Transactionals.execute(transactional, context -> {
+      ProfileDataset profileDataset = getProfileDataset(context);
+      ConfigDataset configDataset = getConfigDataset(context);
+      NamespaceId namespaceId =
+        namespace.equals(PreferencesStore.EMPTY_NAMESPACE) ? NamespaceId.SYSTEM : new NamespaceId(namespace);
+      validateProfileProperties(namespaceId, config.getProperties(), profileDataset);
+      configDataset.createOrUpdate(namespace, PreferencesStore.PREFERENCES_CONFIG_TYPE, config);
+    }, NotFoundException.class, ProfileConflictException.class);
+  }
+
+  private void validateProfileProperties(NamespaceId namespaceId, Map<String, String> propertyMap,
+                                         ProfileDataset profileDataset)
+    throws NotFoundException, ProfileConflictException {
+    Optional<ProfileId> profile = SystemArguments.getProfileIdFromArgs(namespaceId, propertyMap);
+    if (!profile.isPresent()) {
+      return;
+    }
+
+    ProfileId profileId = profile.get();
+    if (profileDataset.getProfile(profileId).getStatus() == ProfileStatus.DISABLED) {
+      throw new ProfileConflictException(String.format("Profile %s in namespace %s is disabled. It cannot be " +
+                                                         "assigned to any programs or schedules",
+                                                       profileId.getProfile(), profileId.getNamespace()), profileId);
+    }
+  }
+
+  private ConfigDataset getConfigDataset(DatasetContext context) {
```

Comment:
can remove this and just use ConfigDataset.get() directily

Reply:
removed